server/server.go

```diff
@@ -147,6 +149,52 @@ func (s *server) ServeFile(ctx context.Context, w http.ResponseWriter, r *http.R
 	})
 }
 
+func (s *server) ServeLog(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	repoName := r.URL.Query().Get(":repo")
+	path := pat.Tail("/log/:repo/", r.URL.Path)
+	offsetStr := r.URL.Query().Get("offset")
+
+	offset := 0
+	if offsetStr != "" {
+		var err error
+		offset, err = strconv.Atoi(offsetStr)
+		if err != nil {
+			http.Error(w, "Invalid offset", 500)
```

Comment:
Try pivoting this to 400, as it's a problem with the user input rather than with the server failing to fetch the log.